src/rules/no-confusing-set-timeout.ts

```diff
@@ -0,0 +1,66 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import { createRule, getNodeName, parseJestFnCall } from './utils';
+
+function isJestSetTimeout(node: TSESTree.Node) {
+  return getNodeName(node) === 'jest.setTimeout';
+}
+
+export default createRule({
+  name: __filename,
+  meta: {
+    docs: {
+      category: 'Best Practices',
+      description: 'Disallow using confusing setTimeout in test',
+      recommended: false,
+    },
+    messages: {
+      globalSetTimeout: '`jest.setTimeout` should be call in `global` scope.',
+      multipleSetTimeouts:
+        'Do not call `jest.setTimeout` multiple times, as only the last call will have an effect.',
```

Comment:
Linting messages shouldn't use full stops because they might appear in sentences constructed by IDEs and such:

```suggestion
      globalSetTimeout: '`jest.setTimeout` should be call in `global` scope',
      multipleSetTimeouts:
        'Do not call `jest.setTimeout` multiple times, as only the last call will have an effect',
```